### DISCUSSION TITLE
How to use URL patterns with downloads.prevent_mixed_content

### DISCUSSION THREAD

**Discussion by AckslD** [qutebrowser/qutebrowser#8381] [Q&A]
I'm struggling a bit to understand how to use URL patterns for the setting `downloads.prevent_mixed_content`. For example I have a URL for which I do want to allow this. Specifically I would want to be able to download the PDF:

http://my.domain.com/pages/my_document.pdf

Without any settings this is prevented with an error. However if I for example add the setting:
```
config.set('downloads.prevent_mixed_content', False, 'http://my.domain.com/pages/my_document.pdf')
```
I can download the PDF. But for some reason this is also true if I write a different domain:
```
config.set('downloads.prevent_mixed_content', False, 'http://my.SOMETHING_ELSE.com/pages/my_document.pdf')
```
On the other hand if I just write:
```
config.set('downloads.prevent_mixed_content', False, 'my.domain.com')
```
it does not work? I'm not understanding why, does anyone have some ideas?

**Comment by The-Compiler** [comment]
Without an actual example and the actual settings being set, all I can say is "can't reproduce :shrug:" really.

- `qutebrowser --temp-basedir https://webdbg.com/test/download/insecure.aspx` and following the first download link -> blocked
- `qutebrowser --temp-basedir ':set -u https://webdbg.com/test/download/insecure.aspx downloads.prevent_mixed_content False' https://webdbg.com/test/download/insecure.aspx` and doing so -> works
- `qutebrowser --temp-basedir ':set -u https://example.com/test/download/insecure.aspx downloads.prevent_mixed_content False' https://webdbg.com/test/download/insecure.aspx` -> blocked again
- `qutebrowser --temp-basedir ':set -u webdbg.com downloads.prevent_mixed_content False' https://webdbg.com/test/download/insecure.aspx` -> works

Note that (like with all other settings) the pattern matches on the page you're visiting, not on the download URL.

**Reply by AckslD** [reply]
Thanks for your message @The-Compiler! My current example is behind my company's VPN, if I come across a public example I will share it here.
